consensus/src/threshold_simplex/types.rs

```diff
@@ -0,0 +1,1210 @@
+            Activity::Nullify(v) => v.view(),
+            Activity::Nullification(v) => v.view(),
+            Activity::Finalize(v) => v.view(),
+            Activity::Finalization(v) => v.view(),
+            Activity::ConflictingNotarize(v) => v.view(),
+            Activity::ConflictingFinalize(v) => v.view(),
+            Activity::NullifyFinalize(v) => v.view(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct ConflictingNotarize<D: Digest> {
+    pub proposal_1: Proposal<D>,
+    pub signature_1: PartialSignature,
+    pub proposal_2: Proposal<D>,
+    pub signature_2: PartialSignature,
+}
+
+impl<D: Digest> ConflictingNotarize<D> {
+    pub fn new(
+        proposal_1: Proposal<D>,
+        signature_1: PartialSignature,
+        proposal_2: Proposal<D>,
+        signature_2: PartialSignature,
+    ) -> Self {
+        ConflictingNotarize {
+            proposal_1,
+            signature_1,
+            proposal_2,
+            signature_2,
+        }
+    }
+
+    pub fn verify(
+        &self,
+        identity: &Poly<Public>,
+        public_key_index: Option<u32>,
+        notarize_namespace: &[u8],
+    ) -> bool {
+        let public_key_index = public_key_index.unwrap_or(self.signature_1.index);
+        let notarize_message_1 = self.proposal_1.encode();
+        let notarize_message_1 = (Some(notarize_namespace), notarize_message_1.as_ref());
+        let notarize_message_2 = self.proposal_2.encode();
+        let notarize_message_2 = (Some(notarize_namespace), notarize_message_2.as_ref());
+        partial_verify_multiple_messages(
+            identity,
+            public_key_index,
+            &[notarize_message_1, notarize_message_2],
+            [&self.signature_1, &self.signature_2],
+        )
+        .is_ok()
+    }
+}
+
+impl<D: Digest> Attributable for ConflictingNotarize<D> {
+    fn signer(&self) -> u32 {
+        self.signature_1.index
+    }
+}
+
+impl<D: Digest> Viewable for ConflictingNotarize<D> {
+    fn view(&self) -> View {
+        self.proposal_1.view()
+    }
+}
+
+impl<D: Digest> Write for ConflictingNotarize<D> {
+    fn write(&self, writer: &mut impl BufMut) {
+        self.proposal_1.write(writer);
+        self.signature_1.write(writer);
+        self.proposal_2.write(writer);
+        self.signature_2.write(writer);
+    }
+}
+
+impl<D: Digest> Read for ConflictingNotarize<D> {
+    fn read_cfg(reader: &mut impl Buf, _: &()) -> Result<Self, Error> {
+        let proposal_1 = Proposal::read(reader)?;
+        let signature_1 = PartialSignature::read(reader)?;
+        let proposal_2 = Proposal::read(reader)?;
+        let signature_2 = PartialSignature::read(reader)?;
+        if proposal_1.view != proposal_2.view {
+            return Err(Error::Invalid("conflicting_notarize", "mismatched views"));
+        }
+        if signature_1.index != signature_2.index {
+            return Err(Error::Invalid(
+                "conflicting_notarize",
```

Comment:
Nit: for consistency the first string should be something like `"consensus::threshold_simplex::ConflictingNotarize"`